pkg/manage/create_docker_compose.go

```diff
@@ -0,0 +1,164 @@
+package manage
+
+import (
+	"context"
+	_ "embed" // Neeed for embed. See Docu of Go 1.16
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"io"
+	"io/fs"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+)
+
+// CreateFileType provides access to os.Create and enables mocking during testing.
+type CreateFileType func(name string) (io.WriteCloser, error)
+
+var CreateFile CreateFileType = func(name string) (io.WriteCloser, error) {
+	return os.Create(name)
+}
+
+// CreateDockerComposeYML creates a docker-compose.yml file in the current working directory
+// using a template. In remote mode it uses the GitHub API to fetch the required commit IDs
+// of all services. Else it uses relative paths to local code as provided in OpenSlides
+// main repository.
+func CreateDockerComposeYML(ctx context.Context, dataPath string, remote bool) error {
+	p := path.Join(dataPath, "docker-compose.yml")
+
+	f, err := CreateFile(p)
+	if err != nil {
+		return fmt.Errorf("creating file `%s`: %w", p, err)
+	}
+	defer f.Close()
+
+	if err := constructDockerComposeYML(ctx, f, remote); err != nil {
+		return fmt.Errorf("writing content to file `%s`: %w", p, err)
+	}
+
+	return nil
+}
+
+//go:embed docker-compose.yml.tpl
+var defaultDockerCompose string
+
+// tplData holds the data used to execute the docker-compose.yml template.
+type tplData struct {
+	Tag                string
+	ExternalHTTPPort   string
+	ExternalManagePort string
+	Service            map[string]string
+}
+
+// Service holds service metadata from GitHub API that are used in the
+// docker-compose.yml template.
+type Service struct {
+	Name    string `json:"name"`
+	HTMLURL string `json:"html_url"`
+}
+
+// constructDockerComposeYML writes the populated template to the given writer.
+// If remote is true it uses GitHub URIs for the build context. Else it uses relative
+// paths to local code as provided in OpenSlides main repository.
+func constructDockerComposeYML(ctx context.Context, w io.Writer, remote bool) error {
+	composeTPL, err := template.New("compose").Parse(defaultDockerCompose)
+	if err != nil {
+		return fmt.Errorf("creating Docker Compose template: %w", err)
+	}
+	composeTPL.Option("missingkey=error")
+
+	const ref = "openslides4-dev"
+
+	var td tplData
+
+	td.Tag = "4.0-dev"
+	td.ExternalHTTPPort = "8000"
+	td.ExternalManagePort = "9008"
+	if err := populateServices(ctx, &td, ref, remote); err != nil {
+		return fmt.Errorf("populating services to template data: %w", err)
+	}
+
+	if err := composeTPL.Execute(w, td); err != nil {
+		return fmt.Errorf("writing Docker Compose file: %w", err)
+	}
+
+	return nil
+}
+
+// populateServices is a small helper function that populates service metadate
+// to the given template data.
+func populateServices(ctx context.Context, td *tplData, ref string, remote bool) error {
+	services, err := getServices(ctx, ref)
```

Comment:
In go, a function should not be called `GetFoo` but just `Foo`: https://golang.org/doc/effective_go#Getters

Reply:
Done.